autotest/gdrivers/ogcapi.py

```diff
@@ -0,0 +1,185 @@
+    for c in chars:
+        text = text.replace(c, "_")
+    return text.replace("_fakeogcapi", "request")
+
+
+class OGCAPIHTTPHandler(BaseHTTPRequestHandler):
+    def log_request(self, code="-", size="-"):
+        pass
+
+    def do_GET(self):
+
+        try:
+
+            request_data_path = os.path.join(
+                BASE_TEST_DATA_PATH, sanitize_url(self.path) + ".http_data"
+            )
+
+            if RECORD:
+
+                with open(request_data_path, "wb+") as fd:
+                    response = requests.get(
+                        "https://demo.pygeoapi.io/stable"
+                        + self.path.replace("/fakeogcapi", ""),
+                        stream=True,
+                    )
+                    content = response.content.replace(
+                        b"https://demo.pygeoapi.io/stable",
+                        (
+                            "http://"
+                            + self.address_string()
+                            + ":"
+                            + str(self.server.server_port)
+                            + "/fakeogcapi"
+                        ).encode("utf8"),
+                    )
+                    data = b"HTTP/2 200 OK\r\n"
+                    for k, v in response.headers.items():
+                        if k == "Content-Length":
+                            data += (
+                                k.encode("utf8")
+                                + b": "
+                                + str(len(content)).encode("utf8")
+                                + b"\r\n"
+                            )
+                        else:
+                            data += (
+                                k.encode("utf8") + b": " + v.encode("utf8") + b"\r\n"
+                            )
+                    data += b"\r\n"
+                    data += content
+                    fd.write(data)
+                    self.wfile.write(data)
+
+            elif self.path.find("/fakeogcapi") != -1:
+
+                with open(request_data_path, "rb+") as fd:
+                    self.wfile.write(fd.read())
+
+            return
+
+        except IOError:
+            pass
+
+        self.send_error(404, "File Not Found: %s" % self.path)
+
+
+###############################################################################
+# Test underlying OGR drivers
+#
+
+
+pytestmark = pytest.mark.require_driver("OGCAPI")
+
+###############################################################################
+@pytest.fixture(autouse=True, scope="module")
+def module_disable_exceptions():
+    with gdaltest.disable_exceptions():
+        yield
+
+
+@pytest.fixture(autouse=True, scope="module")
+def ogr_ogcapi_init():
+    gdaltest.ogcapi_drv = ogr.GetDriverByName("OGCAPI")
+
+
+def test_ogr_ogcapi_fake_ogcapi_server():
+    if gdaltest.ogcapi_drv is None:
+        pytest.skip()
+
+    (process, port) = webserver.launch(handler=OGCAPIHTTPHandler)
+    if port == 0:
+        pytest.skip()
```

Comment:
you'd probably want to move that in a pytest.fixture(autouse=True, scope="module") and do the webserver.server_stop(process, port) there too, like in autotest/ogr/ogr_oapif.py


```suggestion
    (process, port) = webserver.launch(handler=OGCAPIHTTPHandler)
    if port == 0:
        pytest.skip()
```